From patchwork Wed Jul 20 02:17:55 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12923292
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id AE42CC433EF
	for <linux-mm@archiver.kernel.org>; Wed, 20 Jul 2022 02:55:55 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 206006B0073; Tue, 19 Jul 2022 22:55:55 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 1B61C6B0074; Tue, 19 Jul 2022 22:55:55 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 02EEB6B0078; Tue, 19 Jul 2022 22:55:54 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0012.hostedemail.com
 [216.40.44.12])
	by kanga.kvack.org (Postfix) with ESMTP id E971F6B0073
	for <linux-mm@kvack.org>; Tue, 19 Jul 2022 22:55:54 -0400 (EDT)
Received: from smtpin17.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay06.hostedemail.com (Postfix) with ESMTP id B53FCAACED
	for <linux-mm@kvack.org>; Wed, 20 Jul 2022 02:55:54 +0000 (UTC)
X-FDA: 79705963428.17.D874C40
Received: from mx0b-00069f02.pphosted.com (mx0b-00069f02.pphosted.com
 [205.220.177.32])
	by imf17.hostedemail.com (Postfix) with ESMTP id 388E64008E
	for <linux-mm@kvack.org>; Wed, 20 Jul 2022 02:55:54 +0000 (UTC)
Received: from pps.filterd (m0246632.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 26K1lw0C003839;
	Wed, 20 Jul 2022 02:18:33 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=UA7JQTZ/EY25+xzPCgfxRsv7LOWbOF6vDNGzVSDECso=;
 b=EmdKIT4t0g/fgV/s22o0/yGHadGHZv8Imwd/TNWBm6UpyiWpaHf/ngEmCiPQ0sZiEJJT
 VdGBtwqWlJLlQfOqfw3zqRrHE7xdh65rQPncLU9qo+NWN75t24GM7fyW+MAC65DkRI1y
 ukro5I+FPvahSd9d6ejkeBaXq4CP6j6s+r5qfmZOxEOgiAh+wo4ZhR/Irnm2tsF8UU8z
 dUSJHW5K7sxwX84/Jk8dpmgh6QWjCp+l54jkCvmDHdMeJbiS8NzizrsUBaWXcX74ZuzP
 mn5Ah8MlcV4EYSFhhyLwRCdO1qQKzpjGTefLTA+c+WF94RyHr7D6mmkG5xpqC7OTINpF VA==
Received: from iadpaimrmta02.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta02.appoci.oracle.com [147.154.18.20])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3hbmxs83kr-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Wed, 20 Jul 2022 02:18:33 +0000
Received: from pps.filterd
 (iadpaimrmta02.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta02.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 26K1UAlg002689;
	Wed, 20 Jul 2022 02:18:32 GMT
Received: from nam10-bn7-obe.outbound.protection.outlook.com
 (mail-bn7nam10lp2101.outbound.protection.outlook.com [104.47.70.101])
	by iadpaimrmta02.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3hc1mbnuv5-6
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Wed, 20 Jul 2022 02:18:32 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=m2xPxW/buo/a/Yc5jDN1eFyLayOSZHoM4G3HXsxeFSe6GfbLAa2XikhC4PUyhnlSYnkeNzLoWoXWmxwlq6nFKOzwhCT0hDg3DzkTqyv6i4HNdLUm5X0oZ7TWbb2wu6mFSRja73rWM8tosB9/p4o15Qb41VmqPGwC3OuN4Mhg3M67p3sTTHtawBi31tKLLF6pZDzsYUAwxyD4RxjNXHZA9cYOGmTl1AVptl0tbqoQNiko6Mopyedz3xWILGJJeoIZuUem8Np97QZOEgFuPWu0aujLiqks4faOt89jwYvmtfQitLbZHWeQXJVSRyJVsXkNVLvbZUGesMlc/c8DSjTtAw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=UA7JQTZ/EY25+xzPCgfxRsv7LOWbOF6vDNGzVSDECso=;
 b=KdfUrA9/+CTz16DbgOrG6bkd2dFdyaTUZh7VMJ7wtUaZ8t30Kzu/OCbDLqO4NMlYGEq98a4ESaJ8iWZmAi6q312/IItsRxonJNU+HS/jct2y2HJP7ER2ZiRp3861HUiRej2MHFLLre9LQdebp6+nPNj9pyrduwXzKp0qDSCOXtca+qc4G8VJ6X0V/XpO8vH/AjY+9YhZfy3LKbCF0tRQsdKpPZhHyfWyRay+bVkg63NnT83jqz8wA0xkIQvYXEtPYiCz1NBhb5njHWyleXpLXgTL01gnyQnSC0EevETXuPO0jXsfHI1FPx8+dEtPJ7JhAqvcvRzErbdQViuLtdMhDQ==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=UA7JQTZ/EY25+xzPCgfxRsv7LOWbOF6vDNGzVSDECso=;
 b=wehR/LMKblZmHEOfLgcqQW3K/4wwEHydP4emS8t+AbFC/EyDqDNkGiDMErQjvocgj2PGgFlV/owLDptur/6RVuqH7GsreHJ0sabsDMmFn+mO/q9lhRhmCF9Sco2mdvIgtJrLG0g7BCeWESek/h30wyTApZuQZ/Me7WaKw3IsSjQ=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by PH8PR10MB6316.namprd10.prod.outlook.com (2603:10b6:510:1cf::8) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5438.22; Wed, 20 Jul
 2022 02:18:31 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::c4d1:edc3:7d21:7c68]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::c4d1:edc3:7d21:7c68%6]) with mapi id 15.20.5438.024; Wed, 20 Jul 2022
 02:18:31 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>,
        Hugh Dickins <hughd@google.com>
CC: Yu Zhao <yuzhao@google.com>
Subject: [PATCH v12 35/69] x86: remove vma linked list walks
Thread-Topic: [PATCH v12 35/69] x86: remove vma linked list walks
Thread-Index: AQHYm97sFCn8keb6BEuCXyMIjt9bLA==
Date: Wed, 20 Jul 2022 02:17:55 +0000
Message-ID: <20220720021727.17018-36-Liam.Howlett@oracle.com>
References: <20220720021727.17018-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220720021727.17018-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: 62edf9c4-0ffc-4d30-cbeb-08da69f62412
x-ms-traffictypediagnostic: PH8PR10MB6316:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 y3yabNb3I9LAHxafT7dVaTVn81SF5L9+vC9AgD0m43TrgVtNDeNGRoU+5fjhgQ16CUvJ56hsejTPCt8DpEzj5uq9WebfLZHWxJXjhIwFasqOdrwiQkxKBOeofsidDL82F8I6dkEP0RExuiFRyzgCWTbzuacVK5j6FsywNhlr5RmRURxn1tqgFPWn3cqHArzIVepOb2YtSFrI7uq79l2PbETjlGBdtycMVXadK3L8i5CFDwlo7PXOUGvcTTzeg1TAfMfGr6Y5FGwzYA2YExv9vxd7fji9MmGH3MnO9MrDZfI35HFFcSXKKSPh3z2XS2tcdhsF5v2t3Rsj5Fo5fzQqJcoljpMHU46TZ/N1pQ52Xxb3ERZ/sXDcytTnyAtCeRyoHcoiAvumDmFPShrS/riGhWIUYFP/V4P65RJkSYsxjrn+4IGIN6cDXv+93iMJcSd6t3YnXqn5S56up9M3RtApeKUsDfw+z0I/lz8bXw9Y+sMlU7oiog9Y1xKDt1iVxNAaMosUBnvM/Y5qKFoTKNWpiiMAmthKx/oPQnCaVv/nGQ3MlgvEeafKxkhkkVB2bbGRNrpMwmG8+VvaXgHolq5uxcW8EVAg7cFqH9+yHeOmJuc1NDwMDobYSClqAV07Wb9rdoHFlruNDAelWj1Uo3xK4O7CxEiyP7QBRRCmyspCrWdL7oMH0V0UJq68i+nS4euEAaOs944LwMZXtmohXuOO4nXU+f23I+tE7Eo89GRSRKBXawH4OzdWnfxmG3DBKKE8QzEf1OqcGe8IMN7GG2j187AUrRU3xvpGtyxJt3HEt+q3Lg39Ggs49vyRvjOwnDf64LvBI2pqBumxUCn6CogmSqgPs7eT/+q3rowh09fo0uY=
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(346002)(396003)(39860400002)(366004)(376002)(136003)(76116006)(122000001)(44832011)(8936002)(5660300002)(186003)(1076003)(83380400001)(2616005)(4326008)(66446008)(64756008)(36756003)(66946007)(2906002)(8676002)(91956017)(66556008)(316002)(26005)(41300700001)(966005)(66476007)(86362001)(38070700005)(478600001)(6506007)(6486002)(6666004)(71200400001)(110136005)(38100700002)(6512007);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?d4f2fsdbdJh5hXptHq6JiF4?=
	=?iso-8859-1?q?FNFxCm5SaYLanhUsWPiAme0JOzV3iQmkGDmKwiCvC6zD2lUk3qQ9in+oGRsZ?=
	=?iso-8859-1?q?jM2qbbbsLdriCBagsrvLQYUwRLERx7+bXtdUkSrH5U+D+QHIFbcx8/kHAKK4?=
	=?iso-8859-1?q?MaPVSDEJXvbWP8C688auw11B7EhCfZbYGNebgNzPzH5OUfeB2HugELNGrk00?=
	=?iso-8859-1?q?46zCkP1647/FkdIbhjhcibt0BmmlzY0OmZpBnp0NS+UMNA+j9CuuP4fA94Y4?=
	=?iso-8859-1?q?AXtbnB/MRDGpHlvN79uzb+MOO125mTHsN3+H7iUDII8WgJ90rWqjBpgYd6vj?=
	=?iso-8859-1?q?NN8f956APWGRFDH1HUtYEXwgo7wm7o+staMs/B8axmMmp8HQw2VX6OcFZ0Ul?=
	=?iso-8859-1?q?bDT98ch3sJPWVAxBcoxidriL9u3CmPoPSvtdUUY9wD28kcb4FC/pYY65Ng2j?=
	=?iso-8859-1?q?m4QCz4zAov6EFwgpsuXuZGowoDaDkWRgrKL1xLuk0T4UGgrW2gjsJx6o08AG?=
	=?iso-8859-1?q?tL4bvhGWBpvjXfTUAj031am2T92G2qCdq+cMgfQOCJf/PFBLQcBze51dOs6r?=
	=?iso-8859-1?q?9x9Dit0pYIDwGuPbx8ybnXgWFcp9KXwsgrm0fV/bFQQf4zrtqk7/+cUEnLtj?=
	=?iso-8859-1?q?daQZtYzegtudmJJ5bV0m6xT/UPx+2Wr7T1nbsIFeDwxvLIOtzu2uxdgXQXDL?=
	=?iso-8859-1?q?mfThetyki8Y9mNlw6J9B01Jl8V7XZZwW9YB/NFpyOZdbb2nfsHvM7V9gjQFq?=
	=?iso-8859-1?q?7y9n5V25ocRjF14l90Vt6vvHoEbyllOHgKwzuOx6iNlxdeQfc4Gr51AIDrvK?=
	=?iso-8859-1?q?IinVLIQVsSyR9o/kzFG6PsSkOibmBcELt/8xLh5chEIM+OA6YNJi85yZ5ePV?=
	=?iso-8859-1?q?+wbr6mQdohBiwEJbNVjSaOcDAkoJGqhdXiRQNOVn2gBJK7bk0hloudnzRPpv?=
	=?iso-8859-1?q?PQ8F2tTImznIj/RaS0tRpPeerK6D5VxriVR1FNYhMBzRfcDDHoY2DGH+bhXx?=
	=?iso-8859-1?q?nPCq6pKeLC1Kp0sba183IAVHYaBICqbt6srn6LSDZH0eALo6Gp3P4gj9KeOe?=
	=?iso-8859-1?q?bJgW1UWUKbXLzfuGtRBawzYe6Qny4daXocgHWMcs6QZqZPOiIP8G5KH4hlZ4?=
	=?iso-8859-1?q?tlHHPYKj+Ye1cIm0PWELWo75c58+ZG/LUYsk5fIQvFJQRj5cUuR2ClW1flXs?=
	=?iso-8859-1?q?oiNbzqosqwambTvF0ipizAGzSS2WdAhPR2hrgkL+SwlQdtz3wOoSZ8TDOBSH?=
	=?iso-8859-1?q?C+dpGhnxqTrTxesfhzHJdDaXnJQPtyQ92uONQlXWfrzaQN+pJ2AeLU2Yg93m?=
	=?iso-8859-1?q?b6zVWZ4qbM5T5iLPlXG9BX4JXXhBwOnQl/s5DcAZ1vywBEPv4k43IQLcqVwa?=
	=?iso-8859-1?q?cV3K9oi6RnovHhKNs/f6pKLjUl2WkZC0gr+JIi43kr2aXqYbLhXsENd1fOvd?=
	=?iso-8859-1?q?oM8/FRlqdQjZ2P/AxCam6BztwbgtyoCiBk/Yb9O1IZG3UWtfgRfwDM/wjtpu?=
	=?iso-8859-1?q?URhR3uyRohzw2/YaNCITimTaoYh/rdQ5KhMeHlBIXZuAC1WyW3dfA7fiLR/t?=
	=?iso-8859-1?q?WdryIEvwa13iRPS7cNiFm3768fMryvPHxNN3bnHJfB6FBY7rdIePysP5LsqA?=
	=?iso-8859-1?q?tTdMRP5MUSYPvT2tzGGEs1/RvWReQzLOqC+dUYw=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 62edf9c4-0ffc-4d30-cbeb-08da69f62412
X-MS-Exchange-CrossTenant-originalarrivaltime: 20 Jul 2022 02:17:55.4774
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 Tv+af70oPz1/SauLCEBSXufd8W/gNo9hoqrpjX17o4fNtQR/PUd2htaf1NPnh64ca8lqkllTHDWWPB0sUboI2w==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: PH8PR10MB6316
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.883,Hydra:6.0.517,FMLib:17.11.122.1
 definitions=2022-07-19_10,2022-07-19_01,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 malwarescore=0
 bulkscore=0 phishscore=0
 suspectscore=0 mlxlogscore=999 adultscore=0 spamscore=0 mlxscore=0
 classifier=spam adjust=0 reason=mlx scancount=1 engine=8.12.0-2206140000
 definitions=main-2207200008
X-Proofpoint-GUID: 5JghqJnK17_gQZjEPFSSyK8wboWtg0uw
X-Proofpoint-ORIG-GUID: 5JghqJnK17_gQZjEPFSSyK8wboWtg0uw
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1658285754; a=rsa-sha256;
	cv=pass;
	b=bYgxBxhTdhpUEjs1+tGGTxb/ax600DafQrmKSLLxcmD9xtVwxqqe6mWWPb8pHx0qu8EWPA
	TO6nr4bGttsDPXSbQkOxq2QlCtHPCDovqHRK6f3jVFAFvb34873A6o7rwLkSkMYcMvD0Fq
	4/N7nJPayd6eGoSfX7Vz1YkJfhqrZ4Y=
ARC-Authentication-Results: i=2;
	imf17.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=EmdKIT4t;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="wehR/LMK";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=none (imf17.hostedemail.com: domain of liam.howlett@oracle.com has no SPF
 policy when checking 205.220.177.32) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1658285754;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=UA7JQTZ/EY25+xzPCgfxRsv7LOWbOF6vDNGzVSDECso=;
	b=Zt+uwXyo5Xq8gUErZ7ykzReY5EqCv11+Y/DiXCeM+vWNOWGa4hOBj7oBXMHKtB0JLkB+2u
	KA2FcxNPl1FEG0bUij+/mHa+I1Qm+HDXskI8PkFVpF+D5m7w097daUVzST2O7z3HEmAuaV
	+7yw0yZmcIw9oYNirrQh+CfXer3JL4c=
X-Rspam-User: 
X-Rspamd-Server: rspam05
X-Rspamd-Queue-Id: 388E64008E
Authentication-Results: imf17.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=EmdKIT4t;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b="wehR/LMK";
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=none (imf17.hostedemail.com: domain of liam.howlett@oracle.com has no SPF
 policy when checking 205.220.177.32) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Stat-Signature: b5qxchyohisszehkidnqsifigt3udfma
X-HE-Tag: 1658285754-543480
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

Use the VMA iterator instead.

Link: https://lkml.kernel.org/r/20220504011345.662299-20-Liam.Howlett@oracle.com
Link: https://lkml.kernel.org/r/20220621204632.3370049-36-Liam.Howlett@oracle.com
Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Acked-by: Vlastimil Babka <vbabka@suse.cz>
Cc: Catalin Marinas <catalin.marinas@arm.com>
Cc: David Howells <dhowells@redhat.com>
Cc: SeongJae Park <sj@kernel.org>
Cc: Will Deacon <will@kernel.org>
Cc: Davidlohr Bueso <dave@stgolabs.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 arch/x86/entry/vdso/vma.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/arch/x86/entry/vdso/vma.c b/arch/x86/entry/vdso/vma.c
index 111111111..111111111 100644
--- a/arch/x86/entry/vdso/vma.c
+++ b/arch/x86/entry/vdso/vma.c
@@ -127,17 +127,17 @@ int vdso_join_timens(struct task_struct *task, struct time_namespace *ns)
 {
 	struct mm_struct *mm = task->mm;
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	mmap_read_lock(mm);
-
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		unsigned long size = vma->vm_end - vma->vm_start;
 
 		if (vma_is_special_mapping(vma, &vvar_mapping))
 			zap_page_range(vma, vma->vm_start, size);
 	}
-
 	mmap_read_unlock(mm);
+
 	return 0;
 }
 #else
@@ -354,6 +354,7 @@ int map_vdso_once(const struct vdso_image *image, unsigned long addr)
 {
 	struct mm_struct *mm = current->mm;
 	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
 	mmap_write_lock(mm);
 	/*
@@ -363,7 +364,7 @@ int map_vdso_once(const struct vdso_image *image, unsigned long addr)
 	 * We could search vma near context.vdso, but it's a slowpath,
 	 * so let's explicitly check all VMAs to be completely sure.
 	 */
-	for (vma = mm->mmap; vma; vma = vma->vm_next) {
+	for_each_vma(vmi, vma) {
 		if (vma_is_special_mapping(vma, &vdso_mapping) ||
 				vma_is_special_mapping(vma, &vvar_mapping)) {
 			mmap_write_unlock(mm);

